core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -146,6 +148,28 @@ import org.apache.spark.network.sasl.SecretKeyHolder
  *  authentication. Spark will then use that user to compare against the view acls to do
  *  authorization. If not filter is in place the user is generally null and no authorization
  *  can take place.
+ *
+ *  Connection encryption (SSL) configuration is organized hierarchically. The user can configure
+ *  the default SSL settings which will be used for all the supported communication protocols unless
+ *  they are overwritten by protocol specific settings. This way the user can easily provide the
+ *  common settings for all the protocols without disabling the ability to configure each one
+ *  individually.
+ *  The common SSL settings are at `spark.ssl` namespace in Spark configuration, while Akka SSL
+ *  configuration is at `spark.ssl.akka` and HTTP for broadcast and file server SSL configuration
+ *  is at `spark.ssl.fs`. Refer to [[org.apache.spark.SSLOptions]] documentation for the list of
+ *  options that can be specified.
+ *  SecurityManager initializes SSLOptions objects for different protocols separately. SSLOptions
+ *  object parses Spark configuration at a given namespace and builds the common representation
+ *  of SSL settings. SSLOptions is the used to provide protocol specific configuration like TypeSafe
+ *  configuration for Akka or SSLContextFactory for Jetty.
+ *  SSL has to be configured on each node and for each component which is involved in communication
+ *  using the particular protocol. In YARN clusters, the key-store can be prepared on the client
+ *  side and then distributed and used by the executors as the part of the application (it is
+ *  possible because in YARN the user is able to deploy files before the application is started).
```

Comment:
In YARN clusters, the key-store can be prepared on the client side then distributed and used by the executors as the part of the application (YARN allows the user to deploy files before the application is started).